sklearn/pipeline.py

```diff
@@ -33,6 +33,19 @@
 __all__ = ["Pipeline", "FeatureUnion", "make_pipeline", "make_union"]
 
 
+def _final_estimator_has(attr):
+    """Check that final_estimator has `attr`.
+
+    Used together with `avaliable_if` in `Pipeline`."""
+
+    def check(self):
+        # raise original `AttributeError` if `attr` does not exist
+        getattr(self._final_estimator, attr)
+        return True
```

Comment:
As discuss in another PR, I now prefer the following.
I think that I was only missing the comment.

```suggestion
        # returns True or raise an AttributeError if `attr` doe no exist
        return True or getattr(self._final_estimator, attr)
```

Reply:
Actually, discard this solution because if `attr` is a numpy array then this comparison will fail. The current version is better.